container/app/entrypoint.py

```diff
@@ -0,0 +1,67 @@
+#!/venv/bin/python3
+#
+# SPDX-FileCopyrightText: Red Hat, Inc.
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+import argparse
+import json
+import os
+import subprocess
+
+from ovirt_imageio._internal import qemu_img
+from ovirt_imageio._internal import qemu_nbd
+from ovirt_imageio._internal import nbd
+from ovirt_imageio._internal import util
+
+TICKET_PATH = "/app/ticket.json"
+OVIRT_IMAGEIO = "/venv/bin/ovirt-imageio"
+
+
+def main():
+    args = _parse_args()
+    image_info = qemu_img.info(args.image)
+    with util.tmp_dir("imageio-") as base:
+        socket_path = os.path.join(base, "sock")
+
+        ticket_info = _generate_ticket(
+            args.ticket_id, socket_path, image_info["virtual-size"])
+        with open(TICKET_PATH, "w") as f:
+            f.write(json.dumps(ticket_info))
+
+        sock = nbd.UnixAddress(socket_path)
+        with qemu_nbd.run(args.image, image_info["format"], sock):
+            subprocess.run(
+                [OVIRT_IMAGEIO, "--ticket", TICKET_PATH], check=True)
```

Comment:
Nice, we see everything in single function with mostly the same abstraction level.